client/src/components/search/with-instant-search.tsx

```diff
@@ -0,0 +1,208 @@
+import React, { useEffect, useRef } from 'react';
+import type { ReactNode } from 'react';
+import { Location } from '@reach/router';
+import type { WindowLocation } from '@reach/router';
+import { connect } from 'react-redux';
+import { InstantSearch, Configure } from 'react-instantsearch-dom';
+import qs from 'query-string';
+import { navigate } from 'gatsby';
+import Media from 'react-responsive';
+import algoliasearch from 'algoliasearch/lite';
+import { newsIndex } from '../../utils/algolia-locale-setup';
+
+import {
+  isSearchDropdownEnabledSelector,
+  searchQuerySelector,
+  toggleSearchDropdown,
+  updateSearchQuery
+} from './redux';
+import envData from '../../../../config/env.json';
+
+import { createSelector } from 'reselect';
+
+const { algoliaAppId, algoliaAPIKey } = envData;
+
+const DEBOUNCE_TIME = 100;
+
+// If a key is missing, searches will fail, but the client will still render.
+const searchClient =
+  algoliaAppId && algoliaAPIKey
```

Comment:
So, in this case, we only need these to be falsy, not explicitly `null`.  To fix this, could you change config/read-env.js to return `''` for these two properties where it currently returns `null`?

Reply:
It's been done.